File: Python/0011.py
from typing import List

class Solution:
    def maxArea(self, height: List[int]) -> int:
        n = len(height)
        left, right = 0, n - 1
        max_area = 0
        
        while left < right:
            # Calculate current area
            h = min(height[left], height[right])
            width = right - left
            current_area = h * width
            
            # Update max_area if current_area is larger
            if current_area > max_area:
                max_area = current_area
            
            # Move pointers based on which pointer points to shorter line
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        
        return max_area

File: Kotlin/0011.kt
class Solution {
    fun maxArea(height: IntArray): Int {
        var left = 0
        var right = height.size - 1
        var maxArea = 0

        while (left < right) {
            val width = right - left
            val currentHeight = minOf(height[left], height[right])
            val area = width * currentHeight
            maxArea = maxOf(maxArea, area)

            // Move the pointer pointing to the shorter line
            if (height[left] < height[right]) {
                left++
            } else {
                right--
            }
        }

        return maxArea
    }
}
